src/utils/locale.ts

```diff
@@ -95,31 +143,54 @@ export const defaultNumberingSystem =
 export const getSupportedNumberingSystem = (numberingSystem: string): NumberingSystem =>
   isNumberingSystemSupported(numberingSystem) ? numberingSystem : defaultNumberingSystem;
 
-export function getSupportedLocale(locale: string): string {
-  if (locales.indexOf(locale) > -1) {
-    return locale;
-  }
+/**
+ * Gets the locale that best matches the context.
+ *
+ * @param locale – the BCP 47 locale code
+ * @param context - specifies whether the locale code should match in the context of CLDR or T9N (translation)
+ */
+export function getSupportedLocale(locale: string, context: "cldr" | "t9n" = "cldr"): SupportedLocales {
+  const contextualLocales = context === "cldr" ? locales : t9nLocales;
 
   if (!locale) {
     return defaultLocale;
   }
 
+  if (contextualLocales.includes(locale)) {
+    return locale;
+  }
+
   locale = locale.toLowerCase();
 
-  // we support both 'nb' and 'no' (BCP 47) for Norwegian
+  // we support both 'nb' and 'no' (BCP 47) for Norwegian but only `no` has corresponding bundle
   if (locale === "nb") {
     return "no";
   }
 
+  // we use `pt-BR` as it will have the same translations as `pt`, which has no corresponding bundle
+  if (context === "t9n" && locale === "pt") {
+    return "pt-BR";
+  }
+
   if (locale.includes("-")) {
     locale = locale.replace(/(\w+)-(\w+)/, (_match, language, region) => `${language}-${region.toUpperCase()}`);
 
-    if (!locales.includes(locale)) {
+    if (!contextualLocales.includes(locale)) {
       locale = locale.split("-")[0];
     }
   }
 
-  return locales.includes(locale) ? locale : defaultLocale;
+  // we can `zh-CN` as base translation for chinese locales which has no corresponding bundle.
+  if (locale === "zh") {
+    return "zh-CN";
+  }
+
+  if (!contextualLocales.includes(locale)) {
+    console.warn(`Translations for "${locale}" not found or invalid, falling back to english`);
```

Comment:
nitpick: `Translations for "${locale}" are not available, falling back to English (en).`

@geospatialem maybe you could suggest some better warning text?

Reply:
@driskull WDYT about:
> Translations for the "${locale}" locale are not available and will fall back to the default, English (en).

Would like to point developers to a list of currently available locales on our docs site, but would likely be after our 1.0 release.

Reply:
sounds good